### PR TITLE
migrate slick.remotemodel-yahoo to ts

### LINKED ISSUE #12: Migrate slick.remotemodel-yahoo to TS
Migrate slick.remotemodel-yahoo to TypeScript

### PR DESCRIPTION
**Pull request by YPAzevedo**
Closes #12

### CODE REVIEW

**Review by JacobMGEvans — APPROVED** [review]

File: slick.remotemodel-yahoo.ts [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+  /***
+   * A sample AJAX data store implementation.
+   * Right now, it's hooked up to load Hackernews stories, but can
+   * easily be extended to support any JSONP-compatible backend that accepts paging parameters.
+   */
+  function RemoteModel() {
+    // private
+    var PAGESIZE = 10;
+    var data: { [key: number]: any; length: number } = { length: 0 };
+    var h_request: NodeJS.Timeout | null = null;
```
Comment (rbusquet) [inline slick.remotemodel-yahoo.ts:11]: setTimeout in the dom returns a number, not NodeJS.Timeout

Comment (JacobMGEvans) [reply slick.remotemodel-yahoo.ts:11]: I merged, can be fixed in another PR

Comment (JacobMGEvans) [reply slick.remotemodel-yahoo.ts:11]: resolved by #14

Comment (YPAzevedo) [reply slick.remotemodel-yahoo.ts:11]: Hmmm, tried number but the return from setTimeout was NodeJS.Timeout, maybe if i read it from window `window.setTimeout` we also need to fix in slick.remotemodel then, i followed the typing that was there too

Comment (rbusquet) [reply slick.remotemodel-yahoo.ts:11]: moving discussion to #14

**Pull request merged**